app/components/ConnectButton.tsx

```diff
@@ -0,0 +1,124 @@
+import Jazzicon, { jsNumberForAddress } from "react-jazzicon";
+import chainalysisAbi from "~/abi/chainalysis.json";
+import { useAccount, useContractRead, useDisconnect } from "wagmi";
+import { useEffect, useState } from "react";
+import { logger } from "ethers";
+import { ConnectButton } from "@rainbow-me/rainbowkit";
+
+function CustomConnectButton() {
+  const { disconnect } = useDisconnect();
+
+  const [connected, setConnected] = useState(false);
+
+  const account = useAccount({
+    onConnect({ address, connector, isReconnected }) {
+      logger.info(address + "successfully connected");
+      setConnected(true);
+    },
+  });
+
+  const contract_address = "0x40c57923924b5c5c5455c48d93317139addac8fb";
+
+  const { data } = useContractRead({
+    address: contract_address,
+    abi: chainalysisAbi.abi,
+    functionName: "isSanctioned",
+    args: [account.address],
+    enabled: connected,
+  });
+
+  useEffect(() => {
+    if (data === true && connected) {
+      disconnect();
+      setConnected(false);
+      logger.info("address " + account.address + " was sanctioned and therefore disconnected");
+    }
+  }, [data, connected, disconnect, account.address]);
+
+  return (
+    <ConnectButton.Custom>
+      {({ account, chain, openAccountModal, openChainModal, openConnectModal, authenticationStatus, mounted }) => {
+        if (data === true) {
+          authenticationStatus = "unauthenticated";
+        }
+
+        const ready = mounted && authenticationStatus !== "loading";
+        const connected =
+          ready && account && chain && (!authenticationStatus || authenticationStatus === "authenticated");
+
+        return (
+          <div
+            {...(!ready && {
+              "aria-hidden": true,
+              style: {
+                opacity: 0,
+                pointerEvents: "none",
+                userSelect: "none",
+              },
+            })}
+          >
+            {(() => {
+              if (authenticationStatus === "unauthenticated") {
+                return (
+                  <button
```

Comment:
Can we build this on top of maintine's button component so that its all consistent?

Reply:
yeah true